task/gimlet-inspector/src/main.rs

```diff
@@ -0,0 +1,140 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! The Gimlet Inspector provides a deliberately limited set of IPCs over the
+//! network, for extracting diagnostic data from a live system. This is intended
+//! to supplement the more general `dump_agent`.
+
+#![no_std]
+#![no_main]
+
+use drv_gimlet_seq_api::{SeqError, Sequencer};
+use gimlet_inspector_protocol::{
+    QueryV0, Request, SequencerRegistersResponseV0,
+};
+use hubpack::SerializedSize;
+use task_net_api::*;
+use userlib::*;
+
+task_slot!(NET, net);
+task_slot!(SEQ, seq);
+
+// Increase this if a new message requires a larger trailer.
+const MAX_REQ_TRAILER_SIZE: usize = 0;
+
+#[export_name = "main"]
+fn main() -> ! {
+    // Look up our peer task IDs and make clients.
+    let net = Net::from(NET.get_task_id());
+    let seq = Sequencer::from(SEQ.get_task_id());
+
+    const SOCKET: SocketName = SocketName::inspector;
+
+    loop {
+        // These buffers are currently kept kinda small because our protocol
+        // messages are small.
+        let mut rx_data_buf = [0u8; Request::MAX_SIZE + MAX_REQ_TRAILER_SIZE];
+        let mut tx_data_buf = [0u8; 128];
```

Comment:
is there a maximum size constant for the response message type, the way there is for the request?

Reply:
There is, though the response size will be dominated by the trailer. I should extract a constant and document that.

Reply:
Constant extracted.